### DISCUSSION TITLE
CpScreenResponseBehavior -> add "sidebar" functions for the real sidebar

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/cms#13510] [Ideas] [answered]
Your CP has the following structure (`_layout/cp.twig`)
```
┌──────────────────────────────────────────────────┐
│                  #main-content                   │
│   ┌─────┐   ┌──────────────────────┐   ┌─────┐   │
│   │     │   │                      │   │     │   │
│   │  #  │   │                      │   │  #  │   │
│   │  s  │   │                      │   │  d  │   │
│   │  i  │   │                      │   │  e  │   │
│   │  d  │   │       #content       │   │  t  │   │
│   │  e  │   │                      │   │  a  │   │
│   │  b  │   │                      │   │  i  │   │
│   │  a  │   │                      │   │  l  │   │
│   │  r  │   │                      │   │  s  │   │
│   │     │   │                      │   │     │   │
│   └─────┘   └──────────────────────┘   └─────┘   │
│                                                  │
└──────────────────────────────────────────────────┘
sidebar
```

Your CpScreenResponseBehavior has a content and a sidebar function in order to include HTML/Logic there.

```PHP
return $this->asCpScreen()
    ->sidebar('Sidebar HTML')
    ->content('some content HTML');
```

However, in your `CpScreenResponseFormatter` you set the content stuff into the content block, but the sidebar stuff into the details block.
```PHP
'content' => $content,
'details' => $sidebar, // this is actually a bit confusing
```
One would think `content` goes to `content` and `sidebar` to `sidebar`. But `sidebar` does to `details`.
I know this would be a breaking change, so maybe you could consider adjusting this for Craft 5.
In the meantime is there are chance for adding a `navbar` function (or something like that) to place stuff into the `sidebar`

Up until now - I do the following
```PHP
$this->view->on(View::EVENT_BEFORE_RENDER_PAGE_TEMPLATE, static function($event){
    $event->variables['sidebar'] = 'My Sidebar HTML';
});

return $this->asCpScreen()
    ->sidebar('Details HTML')
    ->content('some content HTML');
```
But this kinda feels strange.

Last but not least: I absolutely love your `CpScreenResponse` I used it so much and it's absolutely awesome. It allows me to modify my screens in no time and it makes it easy to work with your Slideouts. This is one of your best changes for me, because it feels like it completes all part of Craft (Slideouts + Elements)

**Comment by brandonkelly** [comment, marked as answer]
It’s awkward because `sidebar` is a little ambiguous right now:

- It means “left sidebar” on full pages
- It means “right sidebar” within slideouts (which don’t have a left sidebar)

CP Screens were added in 4.0 as a common method for defining screens that could be used in both contexts. So we decided to have `sidebar` mean “right sidebar” for those, as that’s the only sidebar that’s actually common across both screen types.

We’ve added support for defining the left sidebar contents on full pages in Craft 4.5 – dubbed `pageSidebar` (#13019). And we’re renaming `sidebar` to `metaSidebar` in Craft 5.0 (#13037). So that will fully resolve this.

**Reply by Anubarak** [reply]
Ah sorry for the trouble... I searched for sidebar in order to make sure this is not a duplicate but I wasn't able to find the other issue.
Thank you 👍
